pkg/metadata/service_metadata.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package metadata
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+)
+
+// ServiceMetadata consists of information about the service and relative links as well
+// as a list of supported/deprecated versions
+type ServiceMetadata struct {
+	Service ServiceDetails `json:"service"`
+	// CRDs to ignore. ACK generator would skip these resources.
+	Versions []ServiceVersion `json:"versions"`
+}
+
+// ServiceDetails contains string identifiers and relevant links for the
+// service
+type ServiceDetails struct {
+	// The full display name for the service. eg. Amazon Elastic Kubernetes
+	// Service
+	FullName string `json:"full_name"`
+	// The short name (abbreviation) for the service. eg. S3
+	ShortName string `json:"short_name"`
+	// The URL of the service's homepage
+	Link string `json:"link"`
+	// The URL of the service's main documentation/user guide
+	Documentation string `json:"documentation"`
+}
+
+// ServiceVersion describes the status of all existing version of the controller
+type ServiceVersion struct {
+	APIVersion string    `json:"api_version"`
+	Status     APIStatus `json:"status"`
+}
```

Comment:
We could also add a boolean field called `IsHub`, to be able to override the hub version. In most of the cases the latest `Available` version is the hub  (prefered storage version). However, according to [k8s deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api) sometimes the hub can be different from the latest version.

Reply:
I am quite averse to the idea of the hub being anything other than the latest generated version - we rely on it being the latest for the common runtime elements to match.